android/java/org/chromium/chrome/browser/BraveAdFreeCalloutDialogFragment.java

```diff
@@ -0,0 +1,114 @@
+/**
+ * Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package org.chromium.chrome.browser;
+
+import android.content.DialogInterface;
+import android.content.res.Configuration;
+import android.graphics.Color;
+import android.graphics.drawable.ColorDrawable;
+import android.os.Bundle;
+import android.util.DisplayMetrics;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.view.WindowManager;
+import android.widget.Button;
+import android.widget.ImageView;
+import android.widget.LinearLayout;
+
+import androidx.fragment.app.DialogFragment;
+
+import org.chromium.chrome.R;
+import org.chromium.chrome.browser.app.BraveActivity;
+import org.chromium.chrome.browser.util.ConfigurationUtils;
+import org.chromium.chrome.browser.util.TabUtils;
+import org.chromium.ui.base.DeviceFormFactor;
+
+public class BraveAdFreeCalloutDialogFragment extends DialogFragment {
```

Comment:
we can possibly create one base class for all dialogfragment and extends it here. we use same extension for many classes  i.e BraveVpnDialogFragment, HighlightDialogFragment, RateDialogFragment